common/src/main/java/com/revenuecat/purchases/common/OfferingParser.kt

```diff
@@ -41,6 +42,7 @@ abstract class OfferingParser {
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     fun createOffering(offeringJson: JSONObject, productsById: Map<String, List<StoreProduct>>): Offering? {
         val offeringIdentifier = offeringJson.getString("identifier")
+        val metadata = offeringJson.optJSONObject("metadata")?.toMap<Any>() ?: emptyMap()
```

Comment:
This is not exactly what we want 🤔 This works fine for a flat dictionary (which is what the API currently supports) but this breaks with a nested array or dictionary because it returns JSONObject and JSONArray. I think we want things to be pure maps and arrays so might need ot do a recursive implementation on this? 🤷‍♂️

Reply:
Yeah, if we want to avoid having those objects, we would need a recursive implementation of this. I don't think we've had to parse an arbitrary nested JSON before this.